### DISCUSSION TITLE
Display hillshade raster using Deck.gl

### DISCUSSION THREAD

**Discussion by lawrencenika** [visgl/deck.gl#9150] [Q&A]
is it achievable? I m using titiler's `/cog/tiles` endpoint with `algorithm=hillshade` and deckgl/react in which tilelayer fetches each image and rendersublayer with bitmaplayer.

**Comment by ibgreen** [comment]
If the hillshading is done by titiler it should just work as the served images will already be shaded? If not your problem is probably not with the hillshader,

**Reply by lawrencenika** [reply]
if the image is already shaded, in that case. should I use TerrainLayer or BitmapLayer?

I tried with TerrraiLayer with the shaded image tile url as `elevationData` while leaving `texture` empty, it then shows weird cubes: 
![image](https://github.com/user-attachments/assets/55b0ac53-1ed8-47a9-9647-f0a20c64d73d)

Using BitmapLayer before this shows line grids which also seems wrong: 
![image](https://github.com/user-attachments/assets/daf9d730-4c5d-4059-b9b0-f17278ab92ee)

Which approach should I dive deeper into?

**Comment by ibgreen** [comment]
You should use the BitmapLayer unless you also have a proper elevatation data source. You may want to check what format titiler is sending, you can do this e.g. in the network tab.

@kylebarron may have some advice.

**Reply by kylebarron** [reply]
I assume it's sending back image tiles, in which case you should use `BitmapLayer` as @ibgreen suggested

**Comment by lawrencenika** [comment]
Thanks both, when i used bitmaplayer, it shows grids why would that be the
case?

**Reply by kylebarron** [reply]
It's probably an issue with your input data that you're passing to titiler
